js/kadai.js

```diff
@@ -0,0 +1,132 @@
+    return this.quiz_difficulty
+  }
+  get question(){
+    return this.quiz_question
+  }
+  get correctAnswer(){
+    return this.quiz_correctAnswer
+  }
+  get choiceList(){
+    return this.quiz_choiceList
+  }
+  shuffle ([...array]) {
+    for (let i = array.length - 1; i >= 0; i--) {
+      const j = Math.floor(Math.random() * (i + 1));
+      [array[i], array[j]] = [array[j], array[i]];
+    }
+    return array;
+  }
+}
+window.addEventListener('DOMContentLoaded', () => {
+  const startButton = document.getElementById('start_button')
+  const titleText = document.getElementById('title')
+  const descriptionText = document.getElementById('description')
+  const genre = document.createElement('p')
+  const difficulty = document.createElement('p')
+  const answerBox = document.createElement('div')
+  const bottomLine = document.getElementById('bottom_line')
+  let quizCount = 0
+  let results = []
+  let numberOfCorrectAnswers = 0
+  let resultLength = 0
+  startButton.addEventListener('click', async () => {
+    startButton.remove()
+    titleText.textContent = '取得中'
+    descriptionText.textContent = '少々お待ち下さい'
+    const responsData = await getQuizData()
+    results = responsData.results
+    resultLength = results.length
+    const quiz_one = new Quiz(results[0].category,results[0].difficulty,results[0].question,results[0].correct_answer,
+      results[0].incorrect_answers);
+    const quiz_two = new Quiz(results[1].category,results[1].difficulty,results[1].question,results[1].correct_answer,
+      results[1].incorrect_answers);
+    const quiz_three = new Quiz(results[2].category,results[2].difficulty,results[2].question,results[2].correct_answer,
+      results[2].incorrect_answers);
+    const quiz_four = new Quiz(results[3].category,results[3].difficulty,results[3].question,results[3].correct_answer,
+      results[3].incorrect_answers);
+    const quiz_five = new Quiz(results[4].category,results[4].difficulty,results[4].question,results[4].correct_answer,
+      results[4].incorrect_answers);
+    const quiz_six = new Quiz(results[5].category,results[5].difficulty,results[5].question,results[5].correct_answer,
+      results[5].incorrect_answers);
+    const quiz_seven = new Quiz(results[6].category,results[6].difficulty,results[6].question,results[6].correct_answer,
+      results[6].incorrect_answers);
+    const quiz_eight = new Quiz(results[7].category,results[7].difficulty,results[7].question,results[7].correct_answer,
+      results[7].incorrect_answers);
+    const quiz_nine = new Quiz(results[8].category,results[8].difficulty,results[8].question,results[8].correct_answer,
+      results[8].incorrect_answers);
+    const quiz_ten = new Quiz(results[9].category,results[9].difficulty,results[9].question,results[9].correct_answer,
+      results[9].incorrect_answers);
+    const quizData = [quiz_one,quiz_two,quiz_three,quiz_four,quiz_five,quiz_six,quiz_seven,quiz_eight,quiz_nine,quiz_ten]
+    startQuiz(quizData)
```

Comment:
`startQuiz` は `async` なので呼び出すときは `await` を使うべきです。
```suggestion
    await startQuiz(quizData)
```